makeflow/src/makeflow.c

```diff
@@ -2385,13 +2377,18 @@ int main(int argc, char *argv[])
 	d->should_read_archive = should_read_archive;
 	d->should_write_to_archive = should_write_to_archive;
 
+	/* Makeflow fails by default if we goto FAILURE_LABEL.
+		This indicates we have correctly initialized. */
+	makeflow_failed_flag = 0;
+
 	makeflow_run(d);
 
 	if(makeflow_failed_flag == 0 && makeflow_nodes_local_waiting_count(d) > 0) {
 		makeflow_failed_flag = 1;
 		debug(D_ERROR, "There are local jobs that could not be run. Usually this means that makeflow did not have enough local resources to run them.");
```

Comment:
while not needed, I would suggest a goto EXIT_WITH_FAILURE here.

Reply:
Agreed. Added. Also used the same D_ERROR stream for the cleanup failure.